src/main/java/soconnect/MainApp.java

```diff
@@ -106,11 +110,23 @@ private ReadOnlySoConnect initSoConnect(Storage storage) {
     private ReadOnlyTodoList initTodoList(Storage storage, ReadOnlySoConnect initialSoConnectData) {
         Optional<ReadOnlyTodoList> todoListOptional;
         try {
-            todoListOptional = storage.readTodoList(initialSoConnectData.getTagList());
+            todoListOptional = storage.readTodoList(initialSoConnectData);
             if (todoListOptional.isEmpty()) {
                 logger.info("Data file not found. Will be starting with a sample TodoList");
             }
-            return todoListOptional.orElseGet(SampleDataUtil::getSampleTodoList);
+            return todoListOptional.orElseGet(() ->
+            {
+                for (int i = 0; i < SampleDataUtil.getSampleTodos().length; i++) {
```

Comment:
You can use `getSampleTagList()` instead of `getSampleTodos()` as all the tags inside the sample todos are already in the sample tag list.

Reply:
But there is still the case where the sample todo is used, but the sample tagList is not used.